monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,133 @@
+
+            test_flight_details = TestFlightDetails(effective_after= effective_after,details = rid_flight_details)
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry= rid_state_data['flight_telemetry'], details_responses = test_flight_details)            
+            test_flight_deserialized = self.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = {'test_id': str(uuid.uuid4()), "requested_flights": requested_flights}        
+
+            all_test_payloads.append({'injection_url':uss['injection_url'], 'injection_payload': test_payload, 'injection_start_time_from_now_secs':uss['start_time_from_now_secs']})        
+        
+        return all_test_payloads
+
+
+class TestHarness():
+    ''' A class to submit Aircraft RID State to the USS test endpoint '''
+
+    def __init__(self, auth_spec:str, auth_url:str):
+        self.auth_spec = auth_spec
+        self.auth_url= auth_url
+        
+    def get_dss_session(self, auth_url:str, auth_spec:str):
+        ''' This method gets a DSS session using the monitoring tools that are provided in the DSS monitoring repository'''
+
+        auth_adapter = make_auth_adapter(auth_spec)
+        s = DSSTestSession(auth_url, auth_adapter)
+    
+        return s
+
+    async def submit_test(self,dss_session, injection_url,  test_payload):
+        print(f"Started: {time.strftime('%X')}")
+        print("Waiting %f seconds" % test_payload['injection_start_time_from_now_secs'])
+        await asyncio.sleep(test_payload['injection_start_time_from_now_secs'])            
+        response = dss_session.put(injection_url, data=test_payload['injection_payload'])
+        print(f"Ended: {time.strftime('%X')}")
+
+        if response.status_code == 200:
+            print("New test with ID %s created" % test_payload['injection_payload']['test_id'])
+        elif response.status_code ==409:
+            print("Test with ID %s already exists" % test_payload['injection_payload']['test_id'])
+        else: 
+            print(response.json())
+
+
+    
+    async def submit_payload_async(self, test_payloads):        
+        ''' This method submits the payload to the injection url by creating a DSSTestSession and then using that session to send the payload '''
+        for test_payload in test_payloads:
+            injection_url = test_payload['injection_url']        
+            auth_sub = urlparse(injection_url).netloc
+            
+            auth_spec_with_sub = self.auth_spec.replace("fake_uss",auth_sub)
+            dss_session = self.get_dss_session(auth_spec= auth_spec_with_sub, auth_url= self.auth_url)
+            dss_session.default_scopes = rid.SCOPE_WRITE
```

Comment:
This is not the scope we want; instead, we want [this one](https://github.com/interuss/dss/blob/fd1d6f4a0a7bc07da88e3dade15e6c8672dd779a/interfaces/automated-testing/rid/injection.yaml#L47)

Reply:
OK I have now created a new variable in the `rid.py` file and referenced it here.